packages/extension-koni-ui/src/Popup/Transaction/variants/Swap/index.tsx

```diff
@@ -1376,18 +1384,43 @@ const Wrapper: React.FC<WrapperProps> = (props: WrapperProps) => {
     }
   }, [goHome, targetAccountProxy]);
 
+  useEffect(() => {
+    fetch(`${BACKEND_API_URL}/swap/fetch-destinations-map`)
+      .then((rawResp) => {
+        if (!rawResp.ok) {
+          throw new Error(`HTTP error! status: ${rawResp.status}`);
+        }
+
+        return rawResp.json() as Promise<{ data: Record<string, string[]>, fromAppVersion: string}>;
+      })
+      .then((resp) => {
+        setPairMap(resp.data);
+        setLoading(false);
+      })
+      .catch((error: Error) => {
+        console.error(`Error while fetching swap destinations: ${error.message}`);
```

Comment:
May stuck at loading screen if can not load data => Need implement error handling if can not load or validate failed